### PR TITLE
Gutenboarding: Add Sidebar

### PR DESCRIPTION
**Pull request by chriskmnds**
#### Changes proposed in this Pull Request

Pulls in the sidebar and a button to show/hide. Sticks closely to the existing sidebar implementation in Gutenberg's post editor. Sidebar content are the selected block's inspector.

Based on https://github.com/WordPress/gutenberg/blob/98c97a718799131968e5bfa17a50858e3508c187/packages/edit-post/src/components/sidebar/index.js

CC @Automattic/luna

### CODE REVIEW

**Comment by ockham** [comment]
With paObgF-BE-p2 in mind, what about replicating `edit-post`'s [directory structure](https://github.com/WordPress/gutenberg/tree/26795cac83cbcd4a78a2045c444d9d30ff2f0590/packages/edit-post/src) more closely? I.e. use a `components/` subfolder for the sidebar etc? (Upstream's `edit-widgets` also [does that](https://github.com/WordPress/gutenberg/tree/26795cac83cbcd4a78a2045c444d9d30ff2f0590/packages/edit-widgets/src) -- hoping to provide some amount of consistency should we ever converge with core's FSE.)

File: client/gutenboarding/components/settings-button/index.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+/**
+ * WordPress dependencies
+ */
+import { IconButton } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+
+/* eslint-disable no-restricted-syntax */
```
Comment (simison) [inline client/gutenboarding/components/settings-button/index.js]: We can remove these eslint-disables

File: client/gutenboarding/components/settings-button/index.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+/**
+ * WordPress dependencies
```
Comment (simison) [inline client/gutenboarding/components/settings-button/index.js]: _WordPress_ → _External_

**Comment by ockham** [comment]
I'll try moving the sidebar button to the header.

**Comment by ockham** [comment]
This is ready for another look. I think my changes (to move the sidebar toggle button into the header) are big enough that it'd make sense to have signoff by someone else :sweat_smile:

**Pull request merged**